src/Sulu/Bundle/SecurityBundle/AccessControl/AccessControlQueryEnhancer.php

```diff
@@ -109,4 +110,57 @@ private function enhanceQueryWithAccessControl(
         $queryBuilder->setParameter('permission', $permission);
         $queryBuilder->setParameter('system', $this->systemStore->getSystem());
     }
+
+    /**
+     * @param array<string|int> $entityIds
+     *
+     * @return array<string|int>
+     */
+    public function findIdsWithGrantedPermissions(
+        ?UserInterface $user,
+        int $permission,
+        string $entityClass,
+        array $entityIds
+    ): array {
```

Comment:
Not sure where we should move this method

Reply:
Maybe the `AccessControlRepository`? as @nnatter mentioned in [this comment](https://github.com/sulu/sulu/pull/6123#discussion_r666194309).

Reply:
do it :)